test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EntityReferenceFunctionalTests.cs

```diff
@@ -26,6 +26,25 @@ public void UseMultipleEscapeSequencesWithRefInKeyAsSegment()
             path.LastSegment.ShouldBeNavigationPropertyLinkSegment(HardCodedTestModel.GetDogMyPeopleNavProp());
         }
 
+        [Fact]
+        public void EntityReferenceCannotAppearAfterMetadata()
+        {
+            PathFunctionalTestsUtil.RunParseErrorPath("$metadata/$ref", ODataErrorStrings.RequestUriProcessor_MustBeLeafSegment("$metadata"));
+        }
+
+        [Fact]
+        public void EntityReferenceCannotAppearAfterSingleton()
+        {
+            PathFunctionalTestsUtil.RunParseErrorPath("Boss/$ref", ODataErrorStrings.PathParser_EntityReferenceNotSupported("Boss"));
+        }

```

Comment:
This is wrong; you should be able to have /$ref after a singleton...

Reply:
Corrected.